packages/openapi3/src/openapi.ts

```diff
@@ -1157,58 +1159,89 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
     parameter: HttpOperationParameter,
     visibility: Visibility
   ) {
-    let defaultToString = false;
     ph.name = parameter.name;
     ph.in = parameter.type;
-    if (parameter.type === "query" || parameter.type === "header") {
-      if (parameter.format === "csv" || parameter.format === "simple") {
-        if (parameter.type === "query") {
-          ph.style = "form";
-          ph.explode = false;
-        } else {
-          ph.style = "simple";
-        }
-      } else if (parameter.format === "multi" || parameter.format === "form") {
-        if (parameter.type === "header") {
-          reportDiagnostic(program, {
-            code: "invalid-format",
-            messageId: "formHeader",
-            format: {
-              value: parameter.format,
-            },
-            target: parameter.param,
-          });
-          defaultToString = true;
-        }
-        ph.style = "form";
-        ph.explode = true;
-      } else if (parameter.format === "ssv") {
-        ph.style = "spaceDelimited";
-        ph.explode = false;
-      } else if (parameter.format === "tsv") {
-        reportDiagnostic(program, {
-          code: "invalid-format",
-          messageId: "tsv",
-          target: parameter.param,
-        });
-        defaultToString = true;
-      } else if (parameter.format === "pipes") {
-        ph.style = "pipeDelimited";
-        ph.explode = false;
-      }
-    }
+
     const paramBase = getOpenAPIParameterBase(parameter.param, visibility);
     if (paramBase) {
       ph = mergeOpenApiParameters(ph, paramBase);
     }
 
-    // Revert unsupported formats to just string schema type
-    if (defaultToString) {
+    const format = mapParameterFormat(parameter);
+    if (format === undefined) {
       ph.schema = {
         type: "string",
       };
-      delete ph.style;
-      delete ph.explode;
+    } else {
+      Object.assign(ph, format);
+    }
+  }
+
+  function mapParameterFormat(
```

Comment:
Split in 2 separate method as anyway the resulting `style/explode` combo are basically mutally exclusive between header and query.